src/util.rs

```diff
@@ -1,20 +1,61 @@
+use std::convert::TryFrom;
 use crate::error::ContractError;
-use cosmwasm_std::{QuerierWrapper, Uint128};
-use provwasm_std::{Marker, MarkerType, ProvenanceQuerier, ProvenanceQuery};
+use cosmwasm_std::{Addr, BankMsg, CosmosMsg, Empty, QuerierWrapper, StdError, StdResult, Uint128};
+use provwasm_std::types::provenance::attribute::v1::{Attribute, AttributeQuerier};
+use provwasm_std::types::provenance::marker::v1::{MarkerAccount, MarkerQuerier};
 use rust_decimal::prelude::Zero;
 use rust_decimal::Decimal;
 use uuid::Uuid;
 
-pub fn is_restricted_marker(querier: &QuerierWrapper<ProvenanceQuery>, denom: String) -> bool {
+pub fn is_restricted_marker(querier: &QuerierWrapper, denom: String) -> bool {
     matches!(
-        ProvenanceQuerier::new(querier).get_marker_by_denom(denom),
-        Ok(Marker {
-            marker_type: MarkerType::Restricted,
+        get_marker(denom.clone(), &MarkerQuerier::new(&querier)),
+        Ok(MarkerAccount {
+            marker_type: 2, // 2 is Restricted
             ..
         })
     )
 }
 
+fn get_marker(id: String, querier: &MarkerQuerier<Empty>) -> StdResult<MarkerAccount> {
+    let response = querier.marker(id)?;
+    if let Some(marker) = response.marker {
+        return if let Ok(account) = MarkerAccount::try_from(marker) {
+            Ok(account)
+        } else {
+            Err(StdError::generic_err("unable to type-cast marker account"))
+        };
+    } else {
+        Err(StdError::generic_err("no marker found for id"))
+    }
+}
+
+// TODO: temporary function, need to find alternative function
```

Comment:
TODO: Add error handling like above

Reply:
Silly question, why does this need error handling like above? 

Above returns `::core::option::Option<crate::shim::Any>` which then needs to be casted to `MarkerAccount` 
https://github.com/provenance-io/provwasm/blob/main/packages/provwasm-std/src/types/provenance/marker/v1.rs#L840

Here returns `::prost::alloc::vec::Vec<Attribute>` directly with no need to cast. 
https://github.com/provenance-io/provwasm/blob/main/packages/provwasm-std/src/types/provenance/attribute/v1.rs#L335

Reply:
not exact same error, just need handle error like not found, or any other error that could happen